File: report/src/main/java/com/vmlens/report/runelementtype/objecthashcodemap/OneThreadIndex.java
package com.vmlens.report.runelementtype.objecthashcodemap;

public class OneThreadIndex implements ThreadIndices {

    private final int threadIndex;

    public OneThreadIndex(int threadIndex) {
        this.threadIndex = threadIndex;
    }

    @Override
    public ThreadIndices addThreadIndex(int newThreadIndex,ObjectHashCodeMap objectHashCodeMap) {
        if(threadIndex == newThreadIndex) {
            return this;
        }
        return new MultipleThreadIndices(objectHashCodeMap.newCode());
    }

    @Override
    public String asUiString() {
        return "";
    }
}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/runelementtype/objecthashcodemap/OneThreadIndex.scala
package com.anarsoft.race.detection.report.element.runelementtype.objecthashcodemap

class OneThreadIndex(private val threadIndex: Int) extends ThreadIndices {
  override def addThreadIndex(newThreadIndex: Int, objectHashCodeMap: ObjectHashCodeMap): ThreadIndices = {
    if (threadIndex == newThreadIndex) return this
    new MultipleThreadIndices(objectHashCodeMap.newCode)
  }

  override def asUiString = ""

  def isSingleThreaded: Boolean = true;
}
